### PR TITLE
DOCS-1420-CE-Early-Preview-3.17.1

### PR DESCRIPTION
**Pull request by bmckercher123**
New version of 3.17.0 docs PR: https://github.com/tigera/docs/pull/667

Product Version(s):
CE 3.17.1

Issue:

Link to docs preview:

SME review:
- [ ] An SME has approved this change. 

DOCS review:
- [x] A member of the docs team has approved this change.

Additional information:

Merge checklist:

- [x] Deploy preview inspected wherever changes were made 
- [x] Build completed successfully
- [x] Test have passed

### CODE REVIEW

**Comment by ctauchen** [comment]
Two things: 
- Would be cleaner to have these as one commit
- the version format in releases.json should be v3.17.1, not just 3.17.1.

File: calico-enterprise_versioned_docs/version-3.17/_includes/release-notes/_v3.17.0-release-notes.mdx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+30 May 2023
+
+### What's new {#whats-new-3.17.0}
+
+:::note
+Due to a tagging error on our part, 3.17.0 was not released. The first release of 3.17 is 3.17.1.
```
Comment (rene-dekker) [inline calico-enterprise_versioned_docs/version-3.17/_includes/release-notes/_v3.17.0-release-notes.mdx]: > Due to a tagging error on our part, 3.17.0 was not released.

Is this what we want to tell the user? What really happened is that we made v3.17.0, followed by v3.17.1 with some fixes. Both are early preview.

Comment (ctauchen) [reply calico-enterprise_versioned_docs/version-3.17/_includes/release-notes/_v3.17.0-release-notes.mdx]: Agree with Rene here. Following same logic as we saw with the 3.16.3 release. Without compelling reason, we shouldn't document and RN for a version we're not releasing. 

I think we should remove the 3.17.0 RN file and the corresponding array in releases.json

Comment (ctauchen) [reply calico-enterprise_versioned_docs/version-3.17/_includes/release-notes/_v3.17.0-release-notes.mdx]: If we do mention the fact that there's no .0 release, we can simply mention the fact without going into details.

Comment (bmckercher123) [reply calico-enterprise_versioned_docs/version-3.17/_includes/release-notes/_v3.17.0-release-notes.mdx]: 3.17.0 has been removed in both places. I kept the note with simplified language (mostly for Support).

**Pull request merged**